src/hotCodeReplace.ts

```diff
@@ -34,19 +38,23 @@ export function initializeHotCodeReplace(context: vscode.ExtensionContext) {
         }
 
         if (customEvent.body.changeType === HcrChangeType.BUILD_COMPLETE) {
-            return vscode.window.withProgress({location: vscode.ProgressLocation.Window}, (progress) => {
-                progress.report({message: "Applying code changes..."});
+            return vscode.window.withProgress({ location: vscode.ProgressLocation.Window }, (progress) => {
+                progress.report({ message: "Applying code changes..." });
                 return customEvent.session.customRequest("redefineClasses");
             });
         }
 
         if (customEvent.body.changeType === HcrChangeType.ERROR || customEvent.body.changeType === HcrChangeType.WARNING) {
             if (!suppressedReasons.has(customEvent.body.message)) {
-                vscode.window.showInformationMessage(`Hot code replace failed - ${customEvent.body.message}`, NOT_SHOW_AGAIN).then((res) => {
-                    if (res === NOT_SHOW_AGAIN) {
-                        suppressedReasons.add(customEvent.body.message);
-                    }
-                });
+                vscode.window.showInformationMessage(
+                    `Hot code replace failed - ${customEvent.body.message}. Would you like to restart the debug session?`,
+                    YES_BUTTON, NO_BUTTON, NEVER_BUTTON).then((res) => {
+                        if (res === NEVER_BUTTON) {

```

Comment:
What does `Never` mean?

Reply:
Shouldn't we always show the error message when `HCR` failed? Refer to `Eclipse`, can it be `suppressed`?

Reply:
We are not exactly match eclipse behavior.  Eclipse can configurate on its settings UI>

Reply:
Not show again for NEVER